numpy/testing/_private/utils.py

```diff
@@ -1794,6 +1794,9 @@ def integer_repr(x):
         return _integer_repr(x, np.int32, np.int32(-2**31))
     elif x.dtype == np.float64:
         return _integer_repr(x, np.int64, np.int64(-2**63))
+    elif x.dtype == np.longdouble:
+        bits = np.finfo(np.longdouble).bits
+        return _integer_repr(x, np.longdouble, np.longdouble(-2**(bits - 1)))
```

Comment:
I have my doubts about double-double here, but am going to ignore them. I'm also not convinced that the subtraction in `_integer_repr` will work correctly for non-integer types as the exponent bits will be treated differently. Is this discussed somewhere?

Reply:
To be more precise, using integers the difference counts ulps, although I would have used offset binary and unsigned integers to handle the negatives. The treatment is IEEE specific. For extended and quad precision, 128 bit integers would be desirable.

Reply:
Would something like this make sense to avoid the use of integers as much as possible?:
```
def nulp_diff(x, y):
    nmant = np.finfo(x.dtype).nmant + 1  # nmant excludes implicit bit
    if x < y:
        x, y = y, x
    return int((x - y) * 2.**53 / 2**np.frexp(x)[1])
```

Reply:
The aim is to count representable floats between two values using the fact that both IEEE floats and integers have the same ordering when interpreted as integers, i.e., the number of float values is the same as the number of integer values even though they represent different numbers. This requires all the bits except for the sign bit, and the sign bit could be used if one is a bit clever. NaNs and Infs do cause trouble in this scheme, so need to be handled separately, although I suspect Infs might fit in, haven't  checked.

Reply:
Can you provide concrete test case examples that will fail here if the machine has sufficient precision? Is the failure worse on this branch because it claims to "support" longdouble now?